### ISSUE TITLE
Comments in source block of transform cause errors with saving and creation

### ISSUE THREAD

**Issue by sjrl** [lf1-io/padl#269]
## Observations
- One common thread appears to be when a comment symbol `#` appears at the beginning of a line when composing a `transform`. It doesn't seem to be an issue if the `#` symbol is also indented.
- If the`#` symbol is present at the beginning of a line in the source block, errors only seem to occur if a `transform` is initialized inline (e.g. `transform(lambda x: x), transform(my_func), MyTransformClass(2)`, etc. If the transform is initialized outside of the block then this problem doesn't seem to occur.

**Comment by sjrl** [comment]
# 1. Comments with `lambda` wrapped transforms
This works 
```python
t = (
    transform(lambda x: x)
    # 
    >> transform(lambda x: x)
)
```
but this does not
```python
t = (
    transform(lambda x: x)
# 
    >> transform(lambda x: x)
)
```
and gives the error message
```python
Traceback (most recent call last):

  File "/Users/sebastianlee/Aleph-One-Gmbh/padl-extensions/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3444, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "/var/folders/y4/wncq_2vx5_j42jklvxtlyjyw0000gn/T/ipykernel_91662/583772831.py", line 4, in <module>
    >> transform(lambda x: x)

  File "/Users/sebastianlee/Aleph-One-Gmbh/padl/padl/wrap.py", line 288, in transform
    return _wrap_lambda(wrappee, ignore_scope)

  File "/Users/sebastianlee/Aleph-One-Gmbh/padl/padl/wrap.py", line 222, in _wrap_lambda
    inner = var2mod.Finder(ast.Lambda).get_source_segments(call)[0][0]

  File "/Users/sebastianlee/Aleph-One-Gmbh/padl/padl/dumptools/var2mod.py", line 42, in get_source_segments
    nodes = self.find(ast.parse(source))

  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/ast.py", line 47, in parse
    return compile(source, filename, mode, flags,

  File "<unknown>", line 1
    >> transform(lambda x:
    ^
SyntaxError: invalid syntax
```

**Comment by sjrl** [comment]
# 2. Comments with `Function` wrapped transforms
Both of these work when initializing the transform
```python
t = (
    padl.Identity()
    # 
    >> transform(np.max)
)
t2 = (
    padl.Identity()
# 
    >> transform(np.max)
)
```
However, when saving only the first one works
```python
t.pd_save('tmp', True)  # works
t2.pd_save('tmp', True)  # Gives error
```
The error for the second save is
```python
Traceback (most recent call last):

  File "/Users/sebastianlee/Aleph-One-Gmbh/padl-extensions/venv/lib/python3.8/site-packages/IPython/core/interactiveshell.py", line 3444, in run_code
    exec(code_obj, self.user_global_ns, self.user_ns)

  File "/var/folders/y4/wncq_2vx5_j42jklvxtlyjyw0000gn/T/ipykernel_91662/3956453609.py", line 1, in <module>
    t.pd_save('tmp', True)

  File "/Users/sebastianlee/Aleph-One-Gmbh/padl/padl/transforms.py", line 231, in pd_save
    code, versions = self._pd_dumps(True, path=path)

  File "/Users/sebastianlee/Aleph-One-Gmbh/padl/padl/transforms.py", line 383, in _pd_dumps
    graph, scopemap = self._pd_build_codegraph(name='_pd_main', scope=scope)

  File "/Users/sebastianlee/Aleph-One-Gmbh/padl/padl/transforms.py", line 1127, in _pd_build_codegraph
    start = self._pd_codegraph_startnode(name)

  File "/Users/sebastianlee/Aleph-One-Gmbh/padl/padl/transforms.py", line 242, in _pd_codegraph_startnode
    start_node = ast.parse(start_source).body[0]

  File "/Applications/Xcode.app/Contents/Developer/Library/Frameworks/Python3.framework/Versions/3.8/lib/python3.8/ast.py", line 47, in parse
    return compile(source, filename, mode, flags,

  File "<unknown>", line 4
    )
    ^
SyntaxError: invalid syntax
```

**Comment by sjrl** [comment]
These work
```python
t = (
    padl.Identity()
#
    >> padl.Identity()
)
t.pd_save('tmp', True)
```
```python
@transform
def func(x):
    return x

t = (
    padl.Identity()
#
    >> func
)
t.pd_save('tmp', True)
```
These do not work
```python
t = (
    padl.Identity()
#
#
    >> padl.Batchify()
)
# Error thrown
```
```python
t = (
    padl.Identity()
#
    >> padl.Batchify()
)
t.pd_save('tmp', True)
# Error thrown
```

**Comment by wuhu** [comment]
Fixed in #272.
The problem was the following:
When creating transforms in line (lambda, functions or classes) the code is being inspected to get the actual call that was used to create the transform (e.g. `transform(lambda x: x)`, `transform(np.max)`, `Batchify()`). To do that we use inspect. However, inspect only gives us the line the call was in, not the columns. To still get the call, we parse the expression containing the call using AST and then compare call nodes in the ast tree with the actual call (more or less). To do that, we first look for the code-block containing the call and then parse it with AST to find the expression containing the call's line (via `dumptools.inspector.get_surrounding_block`). A code block is a block of code with minimum the same indentation - however we need to make sure to ignore comments (which could have a lesser indentation). This check was missing.

**Comment by sjrl** [comment]
Thanks for the explanation!
